### PR TITLE
adds component for updating user status 

### PR DESCRIPTION
**Pull request by TD-17**
updateStatus

### CODE REVIEW

File: app/components/index-buttons.hbs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    <button class="small" type="button" {{on "click" (fn @onUpdateStatus 'ooo')}}><span>Mark yourself as OOO</span></button>
+  {{/if}}   
+  {{#if (eq @status "idle")}}
+    <button class="active" type="button" data-test-button {{on "click" (fn @onUpdateStatus 'active')}}><span data-test-button-text>Change status to ‘Doing a task’</span></button>
+    <button class="small" type="button" {{on "click" (fn @onUpdateStatus 'ooo')}}><span>Mark yourself as OOO</span></button>
+  {{/if}}
+  {{#if (eq @status "ooo")}}
+    <button class="small" type="button" {{on "click" (fn @onUpdateStatus 'active')}}><span>Mark yourself as Active again</span></button>
+  {{/if}}
+</div>
```
Comment (swarajpure) [inline app/components/index-buttons.hbs:24]: We are using BEM for CSS to maintain consistency with class names
You can read up on what's the convention, check `tasks` files if you have doubts (read up first and then see the code, we should get used to learning from docs)

**Review by swarajpure — CHANGES_REQUESTED** [review]
Requesting some changes, mostly around variable naming. Please check!

Good job in there 🥳

File: app/components/index-buttons.hbs (hunk 1 above)
Comment (TD-17) [reply app/components/index-buttons.hbs:24]: Okay

File: app/templates/index.hbs [hunk 2]
```diff
@@ -1 +1,4 @@
-Index page works!
\ No newline at end of file
+<IndexButtons
```
Comment (ankushdharkar) [inline app/templates/index.hbs:1]: Does this name limit us from creating a reusable component in future?

**Review by ankushdharkar — CHANGES_REQUESTED** [review]

File: app/templates/index.hbs (hunk 2 above)
Comment (TD-17) [reply app/templates/index.hbs:1]: Sorry I could not think of any other name at that time! Could I use 'UserStatus' ?

File: app/routes/index.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+const API_BASE_URL = ENV.BASE_API_URL;
+
+export default class TasksRoute extends Route {
+  model = async () => {
+    const response = await fetch(`${API_BASE_URL}/users/self`, {
+      credentials: 'include',
+    });
+    const status = await response.json();
+
+    return status.status;
```
[the remark this answers is not in the document]
Comment (TD-17) [reply app/routes/index.js]: Okay

File: app/controllers/index.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+const BASE_URL = ENV.BASE_API_URL;
+
+export default class IndexController extends Controller {
+  @tracked status = this.model;
+
+  @action async updateStatus(status) {
+    this.status = status;
+    try {
+      const response = await fetch(`${BASE_URL}/users/self`, {
```
Comment (akankshadharkar) [inline app/controllers/index.js:14]: Curious why the fetching of data is happening in the controller and not the route. Also are we not using Ember Data?

File: app/controllers/index.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        method: 'PATCH',
+        body: JSON.stringify({
+          status,
+        }),
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        credentials: 'include',
+      });
+      console.log(response);
```
Comment (ankushdharkar) [inline app/controllers/index.js:24]: No stray console logs in prod please :)

**Review by ankushdharkar — CHANGES_REQUESTED** [review]

File: app/templates/index.hbs (hunk 2 above)
Comment (ankushdharkar) [reply app/templates/index.hbs]: What status of the user does it tell us?

File: app/components/user-status.hbs [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  {{/if}} 
+</div>
+<div class="buttons">
+  {{#if (eq @status "active")}}
+    <button class="buttons__idle" type="button" {{on "click" (fn @onUpdateStatus 'idle')}}><span data-test-button-text>Change status to ‘Idle’</span></button>
+    <button class="buttons--small" type="button" {{on "click" (fn @onUpdateStatus 'ooo')}}><span>Mark yourself as OOO</span></button>
+  {{/if}}   
+  {{#if (eq @status "idle")}}
+    <button class="buttons__active" type="button" data-test-button {{on "click" (fn @onUpdateStatus 'active')}}><span data-test-button-text>Change status to ‘Doing a task’</span></button>
+    <button class="buttons--small" type="button" {{on "click" (fn @onUpdateStatus 'ooo')}}><span>Mark yourself as OOO</span></button>
```
Comment (akankshadharkar) [inline app/components/user-status.hbs:19]: A modifier class is added along side a base class.
Something like `<button class="buttons buttons--small" .... `

You can check out http://getbem.com/naming/ or reach out to me if you have any queries :D

**Review by akankshadharkar — CHANGES_REQUESTED** [review]

File: app/components/user-status.hbs [hunk 7]
```diff
@@ -0,0 +1,24 @@
+<div class="heading">
+  {{#if (eq @status "active")}}
+    <h2 data-test-status>I am doing a task</h2>
+  {{/if}}
+  {{#if (eq @status "idle")}}
+    <h2 data-test-status>I am idle</h2>
+  {{/if}} 
+  {{#if (eq @status "ooo")}}
+    <h2 data-test-status>I am OOO</h2>
+  {{/if}}
```
[the remark this answers is not in the document]
Comment (TD-17) [reply app/components/user-status.hbs:10]: Okay

File: app/components/user-status.hbs [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    <button class="buttons--small" type="button" {{on "click" (fn @onUpdateStatus 'ooo')}}><span>Mark yourself as OOO</span></button>
+  {{/if}}   
+  {{#if (eq @status "idle")}}
+    <button class="buttons__active" type="button" data-test-button {{on "click" (fn @onUpdateStatus 'active')}}><span data-test-button-text>Change status to ‘Doing a task’</span></button>
+    <button class="buttons--small" type="button" {{on "click" (fn @onUpdateStatus 'ooo')}}><span>Mark yourself as OOO</span></button>
+  {{/if}}
+  {{#if (eq @status "ooo")}}
+    <button class="buttons--small" type="button" {{on "click" (fn @onUpdateStatus 'active')}}><span>Mark yourself as Active again</span></button>
+  {{/if}}
+</div>
```
[the remark this answers is not in the document]
Comment (TD-17) [reply app/components/user-status.hbs:24]: Yeah Okay :)

File: app/components/user-status.hbs (hunk 6 above)
Comment (TD-17) [reply app/components/user-status.hbs:19]: Oh yeah right! Thank you :)

File: app/styles/index-buttons.css [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+    background: #ffff;
+}
+
+.buttons--small {
+    font-size: 0.6rem;
+    font-weight: 600;
+    color: #0034a5;
+    border: none;
+    background: #ffff;
+}
```
Comment (sumitd94) [inline app/styles/index-buttons.css:44]: An empty line after this please

File: app/templates/index.hbs [hunk 10]
```diff
@@ -1 +1,4 @@
-Index page works!
\ No newline at end of file
+<UserStatus
+@status={{this.status}}
+@onUpdateStatus={{this.updateStatus}}
+/>
```
Comment (sumitd94) [inline app/templates/index.hbs:4]: An empty line after this please

File: app/styles/index-buttons.css [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+    margin-bottom: 40px;
+    background: #ffff;
+}
+
+.buttons--small {
+    font-size: 0.6rem;
+    font-weight: 600;
+    color: #0034a5;
+    border: none;
+    background: #ffff;
```
Comment (sumitd94) [inline app/styles/index-buttons.css:43]: Also I think the tap width is 4, can we place change it to 2

**Review by sumitd94 — CHANGES_REQUESTED** [review]

File: app/styles/index-buttons.css (hunk 11 above)
Comment (TD-17) [reply app/styles/index-buttons.css:43]: Done

Comment (TD-17) [reply app/styles/index-buttons.css:43]: > Also I think the tap width is 4, can we place change it to 2

Okay

**Comment by swarajpure** [comment]
Fixed in #162

**Pull request closed without merging**